crates/solidity/inputs/language/src/dsl.rs

```diff
@@ -0,0 +1,962 @@
+
+        inline parser ConstructorAttribute = { introduced in "0.4.22" (ModifierInvocation | InternalKeyword | PayableKeyword | PublicKeyword) } ;
+
+        parser ConstructorAttributesList = { introduced in "0.4.22" (ConstructorAttribute +) } ;
+
+        parser ConstructorDefinition = { introduced in "0.4.22" (ConstructorKeyword ParametersDeclaration (ConstructorAttributesList ?) Block) } ;
+
+        parser ContinueStatement = (ContinueKeyword terminated_by Semicolon) ;
+
+        parser ContractDefinition = (
+            { introduced in "0.6.0" (AbstractKeyword ?) } ContractKeyword Identifier (InheritanceSpecifier ?) ((ContractMembersList ?) delimited_by OpenBrace and CloseBrace)
+        ) ;
+
+        inline parser ContractMember = (
+            UsingDirective | FunctionDefinition | ModifierDefinition | StructDefinition | EnumDefinition | EventDefinition |  StateVariableDefinition |
+            { introduced in "0.4.22" ConstructorDefinition } |
+            { introduced in "0.6.0"  (FallbackFunctionDefinition | ReceiveFunctionDefinition) } |
+            { removed in "0.6.0"     UnnamedFunctionDefinition } |
+            { introduced in "0.8.4"  ErrorDefinition } |
+            { introduced in "0.8.8"  UserDefinedValueTypeDefinition }
+        ) ;
+
+        parser ContractMembersList = (ContractMember +) ;
+
+        inline parser ControlStatement = (
+            IfStatement | ForStatement | WhileStatement | DoWhileStatement | ContinueStatement | BreakStatement | DeleteStatement | ReturnStatement | RevertStatement |
+            { introduced in "0.4.21" EmitStatement} |
+            { removed in "0.5.0"     ThrowStatement } |
+            { introduced in "0.6.0"  TryStatement}
+        ) ;
+
+        inline parser DataLocation = (
+            MemoryKeyword | StorageKeyword |
+            { introduced in "0.5.0" CalldataKeyword}
+        ) ;
+
+        parser DeconstructionImport = ((DeconstructionImportSymbolsList delimited_by OpenBrace and CloseBrace) FromKeyword AsciiStringLiteral) ;
+
+        parser DeconstructionImportSymbol = (Identifier ((AsKeyword Identifier) ?)) ;
+
+        parser DeconstructionImportSymbolsList = (DeconstructionImportSymbol separated_by Comma) ;
+
+        parser DeleteStatement = ((DeleteKeyword Expression) terminated_by Semicolon) ;
+
+        parser DoWhileStatement = ((DoKeyword Statement WhileKeyword (Expression delimited_by OpenParen and CloseParen)) terminated_by Semicolon) ;
+
+        inline parser ElementaryType = (
+            BoolKeyword | StringKeyword | AddressType | FixedBytesType | SignedIntegerType | UnsignedIntegerType | SignedFixedType | UnsignedFixedType |
+            { removed in "0.8.0" ByteKeyword}
+        ) ;
+
+        parser EmitStatement = { introduced in "0.5.12" ((EmitKeyword IdentifierPath ArgumentsDeclaration) terminated_by Semicolon) } ;
```

Comment:
I think there is something going on with the Yaml to DSL conversion. For example, `EmitStatement` was introduced in `0.4.21` but here it says `0.5.12`.
What caused that change?

Reply:
No idea. This was a manual change - it must be a result of the general emit working-out.

Fixed.